core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -915,6 +925,20 @@ public Builder customs(ImmutableOpenMap<String, Custom> customs) {
             return this;
         }
 
+        public Builder indexGraveyard(final IndexGraveyard indexGraveyard) {
+            putCustom(IndexGraveyard.TYPE, indexGraveyard);
+            return this;
+        }
+
+        public IndexGraveyard.Builder indexGraveyardBuilder() {
+            final IndexGraveyard graveyard = (IndexGraveyard) getCustom(IndexGraveyard.TYPE);
```

Comment:
I think it will be simpler to just instantiate an empty graveyard in the empty constructor.